uptane-standard.md

```diff
@@ -373,7 +373,7 @@ At a high level, Uptane requires:
     * Snapshot - Indicates images released by the repository at a point in time, via signing metadata about targets metadata
     * Targets - Indicates metadata about images, such as hashes and file sizes
 * A secure way for ECUs to know the time.
-* An in-vehicle client on a primary ECU capable of verifying the signatures on all update metadata, handling all server communication, and downloading updates on behalf of secondary ECUs
+* An in-vehicle client on a primary ECU capable of verifying the signatures on all update metadata, potentially handling all server communication, and downloading updates on behalf of secondary ECUs
```

Comment:
I understand your goal here, but then it still needs to be made clear somewhere that some component has to handle the server communication.

Reply:
Thanks for the feedback, Patrick.

Is your intent to clarify that a "primary" shall be responsible for server communication, or that "some component" shall be responsible for server communication?

I have added my proposed change in latest commit, let me know what you think.

Reply:
My intention was that it was worth specifying that we need a connection to the server somewhere, even if it isn't what we have hitherto called the primary. I think this basically covers that, but I think the language could be better. My preference here would be something like the following:

* An ECU capable of communicating with the server.
* An in-vehicle client on a primary ECU capable of verifying the signatures on all update metadata and downloading updates on behalf of secondary ECUs. The primary ECU MAY be the same ECU that communicates with the server. If they are separate ECUs, there MUST be a secure channel between these ECUs.

Reply:
Thank you for the feedback, Patrick.

I (generally) updated to reflect your recommended change.
I modified 2 main things:
1) We have focused a lot on communication with the server, but in case of update via USB/OBD-II, etc., I believe the essence of the requirement is to 'download images and associated metadata from the Uptane servers'.
This allows an implementer to pass this image & metadata via OTA, USB, OBD-II, etc.
I'd like to hear your thoughts on this approach.

2) Additionally, I omitted the last sentence from the second bullet.
Reason being - I want to be very careful in dictating in-vehicle architecture from this standard.
One theoretical example to justify:
An OEM has a DCM/TCU handling most vehicle<->server communication.
Additionally, this OEM has a domain controller to handle communication from OBD-II.
These two components may be on completely different domains/busses.
Dictating that the DCM & domain-controller (and any other primary included in vehicle) *MUST* have a secure connection between each-other is (in my opinion) **too prescriptive of in-vehicle architecture and should be avoided by the standard**.
I'd like to hear your thoughts (and the thoughts of others) on this.

Thank you again for all of your feedback!

Reply:
Point 1 looks sufficiently addressed to me, thanks!
Point 2 is understandable, although I still think it would be useful to have language about how the network-enabled ECU communicates with the Uptane primary. It's true that we don't want to be prescriptive, and a man-in-the-middle scenario is not very practical here, though. I guess we can leave it as is unless someone has some better ideas.